httpclient/src/main/java/io/wcm/caravan/commons/httpclient/impl/HttpClientFactoryImpl.java

```diff
@@ -126,6 +127,18 @@ public CloseableHttpClient getCloseableWs(URI targetUrl, URI wsAddressingToUri)
     return getFactoryItem(targetUrl, wsAddressingToUri.toString(), targetUrl.getPath(), true).getHttpClient();
   }
 
+  @Override
+  public RequestConfig getDefaultRequestConfig(String targetUrl) {
+    final URI uri = toUri(targetUrl);
+    final String path = uri != null ? uri.getPath() : null;
+    return uri == null ? null : getFactoryItem(uri, null, path, false).getDefaultRequestConfig();
```

Comment:
i think this is not the expected behavior - if the target URL is null is is not null that should be returned, but the default request config of the default item.
but currently the expectation of null/nonnull behavior is not correctly defined for the other API calls as well, we will tackle this in a separate PR.